docs/proposals/object-search.md

```diff
@@ -0,0 +1,160 @@
+- downstream: synchronize the object-search-service updates from the cloud to the edge node.
+- upstream: synchronize the object-search-service updates from the edge to the cloud node.
+
+### Object search service Controller
+![](./images/object-search-controller.png)
+
+The object-search-service controller watches for the updates of object-search-service tasks and the corresponding pods against the K8S API server.
+Updates are categorized below along with the possible actions:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New Object-search-service Created             |Create the cloud/edge worker|
+|Object-search-service Deleted                 | NA. These workers will be deleted by GM.|
+|The corresponding pod created/running/completed/failed                 | Update the status of object-search-service task.|
+
+
+### Downstream Controller
+![](./images/object-search-downstream-controller.png)
+
+The downstream controller watches for object-search-service updates against the K8S API server.
+Updates are categorized below along with the possible actions that the downstream controller can take:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New Object-search-service Created             |Sends the task information to LCs.|
+|Object-search-service Deleted                 | The controller sends the delete event to LCs.|
+
+### Upstream Controller
+![](./images/object-search-upstream-controller.png)
+
+The upstream controller watches for object-search-service task updates from the edge node and applies these updates against the API server in the cloud.
+Updates are categorized below along with the possible actions that the upstream controller can take:
+
+| Update Type                        | Action                                        |
+|-------------------------------     |---------------------------------------------- |
+|Object-search-service Reported State Updated    |  The controller appends the reported status of the object-search-service in the cloud. |
+
+
+### Details of api between GM(cloud) and LC(edge)
+1. GM(downstream controller) syncs the task info to LC:
+    ```go
+    // POST <namespace>/sedna/downstream/objectsearchservices/<name>/insert
+    // body same to the task crd of k8s api, omitted here.
+    ```
+
+1. LC uploads the task status which reported by the worker to GM(upstream controller):
+    ```go
+    // POST <namespace>/sedna/upstream/objectsearchservices/<name>/status
+       
+    // ObjectSearchServiceStatus defines status that send to GlobalManager
+    type ObjectSearchServiceStatus struct {
+    	Phase  string  `json:"phase"`
+    	Status string  `json:"status"`
+    	Output *Output `json:"output"`
```

Comment:
need more elaboration about this ObjectSearchServiceStatus, e.g.:
1. `Output`: Models is needed?
2. `TaskInfo`: only `InferenceNumber`?
3. etc.

Reply:
The "ObjectSearchServiceStatus " and "ObjectTrackingServiceStatus " are updated.